src/wms/src/test/java/org/geoserver/wms/legendgraphic/JSONLegendGraphicOutputFormatTest.java

```diff
@@ -1841,4 +1844,147 @@ private GetLegendGraphicRequest getRequest(FeatureType layer, Style style) {
         req.setFormat(JSONFormat);
         return req;
     }
+
+    @Test
+    public void testLayerGroupStyleSINGLE() throws Exception {
+        Catalog catalog = getCatalog();
+        LayerGroupInfo group = null;
+        try {
+            createLakesPlacesLayerGroup(
+                    catalog, "lakes_and_places_group", LayerGroupInfo.Mode.SINGLE, null);
+            group = catalog.getLayerGroupByName("lakes_and_places_group");
+            LayerGroupStyle groupStyle = new LayerGroupStyle();
+            StyleInfo styleName = new StyleInfoImpl(getCatalog());
+            styleName.setName("nature-style");
+            groupStyle.setName(styleName);
+            groupStyle.getLayers().add(getCatalog().getLayerByName("cite:Forests"));
+            groupStyle.getLayers().add(getCatalog().getLayerByName("cite:Lakes"));
+            groupStyle.getStyles().add(null);
+            groupStyle.getStyles().add(null);
+            group.getLayerGroupStyles().add(groupStyle);
+            catalog.save(group);
+            String url =
+                    "wms?LAYER="
+                            + group.getName()
+                            + "&STYLE=nature-style&FORMAT=application/json"
+                            + "&SERVICE=WMS&REQUEST=GetLegendGraphic&VERSION=1.0.0&WIDTH=20&HEIGHT=20";
+            JSONObject legend = (JSONObject) getAsJSON(url);
+            JSONObject forest = legend.getJSONArray("Legend").getJSONObject(0);
+            JSONObject lakes = legend.getJSONArray("Legend").getJSONObject(1);
+            assertEquals("Forests", forest.get("layerName"));
+            assertEquals("Lakes", lakes.get("layerName"));
+        } finally {
+            if (group != null) {
+                catalog.remove(group);
+            }
+        }
+    }
+
+    @Test
+    public void testLayerGroupStyleOPAQUE() throws Exception {
+        Catalog catalog = getCatalog();
+        LayerGroupInfo group = null;
+        try {
+            createLakesPlacesLayerGroup(
+                    catalog, "lakes_and_places_group", LayerGroupInfo.Mode.OPAQUE_CONTAINER, null);
+            group = catalog.getLayerGroupByName("lakes_and_places_group");
+            LayerGroupStyle groupStyle = new LayerGroupStyle();
```

Comment:
Some of this setup code seems to be copy/pasted. Try to have build methods instead with few params to handle the differences.

Reply:
I see this repeated in other tests too, see what you can do about avoiding it.

Reply:
fixed